lib/Auth/Source/MicrosoftHybridAuth.php

```diff
@@ -48,10 +48,10 @@ protected function postFinalStep(AccessToken $accessToken, AbstractProvider $pro
         $idTokenData = $this->extraIdTokenAttributes($accessToken->getValues()['id_token']);
         $prefix = $this->getAttributePrefix();
 
-        if (array_key_exists('email', $idTokenData)) {
+        if (array_key_exists('email', $idTokenData) && !array_key_exists($prefix . 'mail' , $state['Attributes'])) {
             $state['Attributes'][$prefix . 'mail'] = [$idTokenData['email']];
         }
-        if (array_key_exists('name', $idTokenData)) {
+        if (array_key_exists('name', $idTokenData) && !array_key_exists($prefix . 'name' , $state['Attributes'])) {
```

Comment:
I didn't understand the motivation for this change. I'd prefer not to make it - we validated the existing code with a mix of live, converged and azure ad apps  x  social, guest and azure ad accounts and I don't want to redo that validation :)  I recall that MS was _weird_ in how attributes were returned in some of those situation

Reply:
I returned it to its original state.